### PR TITLE
(#52) Run integration tests in CI.

### LINKED ISSUE #52: Enabler: Run integration tests in CI build
# Run integration tests as part of a GitHub actions workflow.

[Now that we have integration tests](https://github.com/NCIOCPL/sitewide-search-api/issues/48), it's time to start running them as part of every pull request.

> **ESTIMATE** 1

### Prerequisites
- [x] Issue #48

### Sub-Tasks
- [ ] [Change to using v7.9.2 of Elasticsearch](https://github.com/NCIOCPL/sitewide-search-api/blob/develop/.github/workflows/workflow.yml#L125)
- [ ] Load [the synonym list](https://github.com/NCIOCPL/sitewide-search-api/blob/develop/integration-tests/docker-sws-api/elasticsearch/synonym.txt) into Elasticsearch.
- [ ] Uncomment [the lines which load the test data into elasticsearch](https://github.com/NCIOCPL/sitewide-search-api/blob/develop/.github/workflows/workflow.yml#L148-L149).
- [ ] The steps which [execute the tests and verify their success](https://github.com/NCIOCPL/sitewide-search-api/blob/develop/.github/workflows/workflow.yml#L185-L211). 

### Notes

Loading the synonym list is a little tricky. The synonym list needs to be loaded into the elasticsearch configuration directory. When running Docker locally, the file is installed via a file-specific volume mapping. In GitHub actions, services (which are Docker containers) are set up before the workflow steps are executed. Retrieving files (including the synonym list) is one of the first workflow steps.

A potential approach to workaround this limitation is to create a volume mapping and copy the synonym list into it after the files have been retrieved.  An [Elasticsearch blog post about synonyms](https://www.elastic.co/blog/boosting-the-power-of-elasticsearch-with-synonyms) mentions a new API endpoint (`POST /synonym_test/_reload_search_analyzers`) for reloading the analyzers after modifying the synonym list. This might be accomplished via curl. 

The article also mentions that queries with filters which use synonyms should be marked as `updatable`.  This has implications for the mapping file which need to be verified to be certain the search will still behave identically to production. (Otherwise, what's the point?)

### PR DESCRIPTION
**Pull request by blairl-nih**
Closes #52 

TL;DR -- Integration tests weren't being run in CI.  Fixing #95 included adding additional tests and supporting data.  This caused a handful of tests to fail because they were expecting `X` number of results and were instead finding between `X + 2` and `X + 4` results.

This PR fixes both the failing tests and the lack of running them.

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -143,16 +136,22 @@ jobs:
...
-          ELASTIC_SEARCH_HOST: http://localhost:${{ job.services.elasticsearch.ports[9200] }}
+          ELASTIC_SEARCH_HOST: http://localhost:9200
         run: |
-              ## Create test output and API logging location
-              mkdir -p integration-tests/target
+            # Sitewide search uses a synonyms list which is placed directly into the Elasticsearch config
+            # directory. It's not possible to do this with a service container as they are created before the job
+            # has performed its checkout step. We therefore have to manage Docker directly.
+            # Run only elasticsearch.
+            docker compose --file integration-tests/docker-sws-api/docker-compose.yml up --detach elasticsearch
```
Comment (bryanpizzillo) [inline .github/workflows/workflow.yml:145]: This should be its own workflow step to start docker. Then there should be a separate step to stop it cleanly. It should stop even if tests fail.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]

File: .github/workflows/workflow.yml (hunk 1 above)
Comment (blairlearn) [reply .github/workflows/workflow.yml:143]: Seems reasonable.

**Comment by blairlearn** [comment]
Folded into PR #101
